bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,483 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link SecondGenerationHandler} is responsible for handling commands, which are
+ * sent to one of the channels, regarded to second generation part of the binding.
+ *
+ * @author Christian Schneider - Initial contribution
+ * @author Christoph Weitkamp - Incorporated new QuantityType (Units of Measurement)
+ * @author Örjan Backsell - Redesigned regarding Piko1020, Piko New Generation
+ */
+// @NonNullByDefault
+public class SecondGenerationHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SecondGenerationHandler.class);
+
+    private HttpClient httpClient;
+
+    // @Nullable
+    private SecondGenerationConfiguration config;
+
+    @SuppressWarnings("unused")
+    // @Nullable
+    private SecondGenerationBindingConstants configurationConfig;
+
+    private final List<SecondGenerationChannelConfig> channelConfigs = new ArrayList<>();
+    private final List<SecondGenerationChannelConfig> channelConfigsExt = new ArrayList<>();
+    private final List<SecondGenerationChannelConfig> channelConfigsExtExt = new ArrayList<>();
+
+    public SecondGenerationHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+
+    }
+
+    // @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String url = config.url.toString();
+        String username = config.username;
+        String password = config.password;
+        String valueConfiguration = "";
+
+        if (command instanceof RefreshType) {
+            logger.debug("Handle command for {} on channel {}: {}", thing.getUID(), channelUID, command);
+        }
+
+        if (command == RefreshType.REFRESH) {
+            return;
+        }
+
+        if (channelUID.getId().equals("chargeTimeEnd")) {
```

Comment:
Instead of a series of if statements for channel names, it is better to a switch statement for this.
```suggestion
        switch(channelUID.getId()){
            case SecondGenerationConfigurationBindingConstants.CHANNEL_CHARGETIMEEND:
```
Also, please use your channel constants where possible.